bin/extract_api_surface.py

```
@@ -0,0 +1,329 @@
+    # Check for __all__ first - if defined, it controls the public API
+    for node in ast.iter_child_nodes(tree):
+        if isinstance(node, ast.Assign):
+            for target in node.targets:
+                if isinstance(target, ast.Name) and target.id == "__all__":
+                    if isinstance(node.value, (ast.List, ast.Tuple)):
+                        all_exports = []
+                        for elt in node.value.elts:
+                            if isinstance(elt, ast.Constant) and isinstance(
+                                elt.value, str
+                            ):
+                                all_exports.append(elt.value)
```

Ruff (0.15.7)

[warning] Use a list comprehension to create a transformed list

(PERF401)